### ISSUE TITLE
Feature Request DESIGNATE: Please add target coordinates to BRA report and Laser code

### ISSUE THREAD

**Issue by Ramsay58** [FlightControl-Master/MOOSE#675]
**Preamble**
DCS has very poor pre-flight mission planning compared to Falcon BMS, I'm trying to use MOOSE's A2G Task Planner and Designate menus to identify, plan and brief mission targets/threats prior to taxi in a MP environment.
![designation_area_list_screen_170831_161308](https://user-images.githubusercontent.com/31033413/29938633-579d1f70-8e81-11e7-91f9-2261d18ec8b0.jpg)

**Adding Coordinates**
When marking targets (areas) on the F10 map, I need to move around the map and zoom in and out to mark target/threat areas using BRA from our starting AB.
![desigation_using_bra_to_mark_targets_on_f10_map_for_preflight_brief_screen_170831_161631](https://user-images.githubusercontent.com/31033413/29933710-8604fea2-8e70-11e7-8135-ae50a781f06f.jpg)

While I can ask players to mark their own targets (and hence split the work load), it'd be better if the Designate Radio menu included coordinates so we could focus on the target area.
![designation_area_list_f10_map_screen_170831_162613](https://user-images.githubusercontent.com/31033413/29936115-efdbd226-8e78-11e7-842f-8377707c130c.jpg)

**Laser Code**
Unfortunately you can't see which laser code will be used until after you ask for it 
![designation_bra_and_type_f10_map_annotation_screen_170831_163408](https://user-images.githubusercontent.com/31033413/29933715-91590bcc-8e70-11e7-89a6-838310e51263.jpg)

... and are back in the cockpit view.
![designation_laser_code_screen_170831_175453](https://user-images.githubusercontent.com/31033413/29935275-155ed686-8e76-11e7-936a-d2f9c64ca440.jpg)

Perhaps the Lase menu item could include the "laser code", in the same way smoke includes the colour in it's decription i.e. "F2. Lase targets"  becomes "F2. Lase targets (1688).

**Targets in Area - Report Page**
It'd be good if we could see what Target(s) were in the area using the same menu - perhaps a "Area report" page with an "F11 previous menu" option, as the "Status" option is a timed "cockpit view"  message that fades and requires the radio F10 Other/Command Centre/Mission/Designation... chain to be opened again.

Note: Task Reports have the same design issue, looking at Planned tasks, etc. before joining a task requires opening the radio several times, rather than using F11 to get to the previous menu.

**Comment - CTLD JTAC** 
CTLD JTAC works well, so you might want to note the info it provides to players
![ctld_lasing_target_giving_coordinates_screen_170829_012827](https://user-images.githubusercontent.com/31033413/29937094-59c43928-8e7c-11e7-8796-c23b6e95bccc.jpg)

But unlike MOOSE, CTLD only reports on the first target each JTAC is tracking, so can't be used in a pre-flight brief to plan for subsequent  targets or known threats.

**Comment by thebgpikester** [comment]
Change your player settings to the coordinate of choice, coords then appear, default is BRA. The laser code depends on the mission design, so you can choose how you want it, also being tweaked now. The targets in area are counted in the mission breifing. Detection can explain multiple targets, but yeah its done in order, there is a priority system that is optional, but no granularity. It does say however, how many targets in the mission.

As for the menu system, yeah it's deep, im not a fan, but its a lot of info to get via an F1-10 system, so unless there is a really good design overhaul that's recommended specifically, we can;t discuss what is better.

**Comment by FlightControl-Master** [comment]
Thanks @Ramsay58 and @thebgpikester for your request and clarifications.
There are a few elements requested here...

1. Task planning.
Task planning should be possible only with TASKING A2G menus. You don't need designate.
You can ask for various target reports in A2G tasking.
The Planned Tasks report provides you with a comprehensive report sorted per distance.
The format of the coordinates can be selected using the settings A2G menu.
What prevents you from using that report?

2. Target marking:
Is only valid for stationary targets, not for moving ones.
MOOSE could place the marks for you on the map.
Not sure if I can move them.
Let me play with that tomorrow.
Moose continuously informs you of the target location you have assigned.
Targets can move and coordinates may change per location.

3. Lase targets and laser code.
Codes are selected automatically from a list.
I've added in the latest implementation a method to request lases with specific laser codes.
:AddMenuLaserCode( 1680, "Lase for A-10A (%d)" ) will do what you want.
It will add a specific lase menu with a specific code. A player can select and the lase will be adapted.
(That's how it should work).

4. DESIGNATE does more. It lases **multiple targets**. With different laser codes, by different recce.
Does DESIGNATE need to show the coordinates of the targets it is lasing in the format selected in A2G settings?

**Comment by Ramsay58** [comment]
Feedback @FlightControl-Master 

**1. Task Planning ...**
> What prevents you from using that report?

The 20s timeout and the fact it can't be viewed in the F10 view.

However, I've found I was making pre-flight Task Planning harder than needed as I hadn't picked up on (or forgot) that tasks and detection areas used the same sufix (job number) i.e CAS.005 = Area.5
![tasks_and_designate_use_same_postscripts_screen_170901_133717](https://user-images.githubusercontent.com/31033413/29971738-48d84a64-8f21-11e7-97d9-459ff6d787ae.jpg)

as I was using LL DDM and the Designate radio menu lists Area.job_number and BRA, with out any target info.
![task_and_designate_ll_ddm_sidebyside_screen_170901_135841](https://user-images.githubusercontent.com/31033413/29971917-025a16ca-8f22-11e7-8450-73a83628c0bd.jpg)

At work we used job numbers as a prefixes i.e. 004.CAS, 004.AREA, 004.SEAD so that might be part of the reason I missed it. I'll need to re-watch the Detection and Tasking videos to see how moose deals with dropped and reacquired targets (or just fly a bit more). 

**2. Target marking ...**
>Is only valid for stationary targets, not for moving ones.
MOOSE could place the marks for you on the map.

Placing a Tactical Marker might be an option for joined tasks but might be seen as a cheat/virtual data link if used in flight. OTOH it would be great time saver. I wouldn't want auto placement or the map markers updating in real time without user input, as DCS has a "fog of war" option for that type of game play.

If F10 Map marking is a possible MOOSE addition, it should definitely be included in the framework as AFAIK, there is no way for a mission designer to add Tactical Markers in the ME? 

**3. Lase targets and laser code ...**
>I've added in the latest implementation a method to request lases with specific laser codes.
:AddMenuLaserCode( 1680, "Lase for A-10A (%d)" ) will do what you want.

Great stuff - it worked great, though I had expected it to be a sub menu of "Lase targets"
![designate_bra_lase_screen_170901_161800](https://user-images.githubusercontent.com/31033413/29977120-3a52b6e2-8f34-11e7-8961-ecbdb08ce99d.jpg)
What happens if there are more than 2 laser codes or is it WIP?

**4. DESIGNATE does more ...**
> It lases multiple targets. With different laser codes, by different recce.
Does DESIGNATE need to show the coordinates of the targets it is lasing in the format selected in A2G settings?

Yes, if it used independently of the TASK_A2G_DISPATCHER Tasking (or it needs it's own "JTAC 9-line" report). 

If "job numbers" always match between Tasks and Designation Areas (CAS.xxx -> Area.xxx), designate coordinate format is less of an issue as Designate becomes a "lase this", "smoke that" function and TASK_A2G_DISPATCHER can provide a most of a "JTAC 9-line" report

**_I would however like BRA, LL and MGRS in single report, rather than swapping user prefrence as I navigate using BRA and map read using LL or MGRS._**

The closest MOOSE has is the Assigned Task Report
![assignedtaskreport_has_metric_and_bra_despite_imperial_and_ll_ddm_preference_screen_170901_161651](https://user-images.githubusercontent.com/31033413/29977021-e69a8f0c-8f33-11e7-94f3-7eede5d489f7.jpg)
but it doesn't respect the Imperial/Metric user preference or include LL and MGRS coordinates.

Drilling down the Designate menu just gives BRA and height
![designate_bra_lase_screen_170901_161800](https://user-images.githubusercontent.com/31033413/29978166-ea9ca73a-8f37-11e7-84a3-634b676d4c5d.jpg)

and the Designate Status Reports BRA, Height, Threat and Target ID
![designate_status_report_screen_170901_161828](https://user-images.githubusercontent.com/31033413/29978452-3b6097ca-8f39-11e7-8483-a5dc92ea0b86.jpg)

The Status Report already spills onto a 2nd line, so perhaps LL and MGRS coordinates could be added there.

**Comment by FlightControl-Master** [comment]
@Ramsay58 My feedback on your feedback :-)

**1. Task Planning ...**

_as I was using LL DDM and the Designate radio menu lists Area.job_number and BRA, with out any target info._

Both DESIGNATE and A2G should provide you with the information in the format as chosen in the settings. If DESIGNATE displays in BRA and A2G tasking in LL, then that is an issue I need to resolve!

**2. Target marking ...**

OK. So no markers... But then again, I can't get all of the coordinates in the menu packed. It has an impact on performance ....

What if I make it as a SETTING where the players can put markers on and off?
Also, I will look into the possibility to remove the markers once the player is airborne.

You can't place markers in the ME, no.

**3. Lase targets and laser code ...**

_What happens if there are more than 2 laser codes or is it WIP?_

DESIGNATE allows you to put as many fixed laser codes as you want, and put additional menus for those. You use the method :AddMenuLaserCode() for that. (http://flightcontrol-master.github.io/MOOSE/Documentation/Designate.html##(DESIGNATE).AddMenuLaserCode)

All the rest of the laser codes are randomly generated. So when the menu option "Lase Targets" is used, it will pick random laser codes from a random generated list and report those. I need to change the name of it, because it confused you.

**4. DESIGNATE does more ...**

_I would however like BRA, LL and MGRS in single report, rather than swapping user prefrence as I navigate using BRA and map read using LL or MGRS._

I could add a SETTING option to show coordinates in the 3 modes (if possible).
Is that an idea? But beware of overloaded information then...
I don't understand why you use BRA and then LL and MGRS...

And I need to rework a bit the DESIGNATE menu coordinates displays and menu...

FC

**Comment by Ramsay58** [comment]
Thanks for taking the time to read my comments. A little feedback.

**2. Target Marking...**
_But then again, I can't get all of the coordinates in the menu packed. It has an impact on performance ...._

I was thinking a "Mark Task Location on Map" option in "Join Task" or Status menu but not if it causes issues, I thought you'd be able to read a unit/area position for plotting tasks (if not tracking the data internally). Tactical Marking is "nice to have" but the overall MOOSE Framework and performance is more so.

_Also, I will look into the possibility to remove the markers once the player is airborne._

I use the F10 Map as I would a paper map, so I'd consider their disappearance (or automatic appearance) unrealistic, stopping update/additions after being airborne is fine but I'd be worried about causing CA or Heli players problems, so would leave it to mission designers.

I'm a great believer in "KISS" (Keep It Simple Stupid) - I make detailed and complicated spider diagrams, then change "hats" - and prune aggressively, so what's left is the core needed (plus some no cost extras).

**4. Designate does more ...**
_I could add a SETTING option to show coordinates in the 3 modes (if possible).
Is that an idea? But beware of overloaded information then..._

It wouldn't be to guide a player/aircraft to a task/area/target (usually BRA)

But for the detailed Task Reports i.e. 

> Task: CAS.005 - <Planned> - Detailed Report
> - Players
> 
> - Coordinates
> BRA, 114° for 41.88 miles at 0 ft
> BULLS, 292° for 52.58 miles
> **LL DDM, 44 41.750' N   40 01.780' E**
> **MGRS, 37T EK 81818 49428**
> - Targets: 1 of Ural-375 ZU-23
> - Threat: [******]

From the info, I read what I need and disregard the rest. 

_I don't understand why you use BRA and then LL and MGRS..._

**Navigation = BRA** - fly 114° for 41.88 miles, easy. LL/MGRS requires knowing my own location and plotting on the F10 map - not easy once airborne without an autopilot). It's a little unrealistic but I like to think IRL there'd be GCI BRA for the COIN op's I model.

**Where is the target = LL or MGRS** - we don't always know exactly where we are once airborne i.e. night, cloud, terrain, half-way between way points in an old aircraft or a player is just lost (I have "own position" on the F10 map disabled). So BRA tells us which direction to fly but not exactly where the target is on the ground i.e. is it near an airbase or a town, in the bottom of a valley or on the ridge line, etc.

LL or MGRS can be entered into A-10C, Ka-50, M-2000C to create a 'target' way point to help find the target without the need to call for smoke, etc.

The mission I'm running is a simple training one ATM but the hope is to expand it to spawn random threats/targets and add an element of ATO so that a player CAS/BAI might have AI CAP and SEAD support (just a couple of aircraft, here and there to convey a sense of being part of something bigger, while not killing the frame rate/causing lag).

**Comment by FlightControl-Master** [comment]
Okay lets come to a conclusion on the requirements ... 

**1. Task Planning ...**

I think you agree (as you haven't reacted) that the A2G settings need to:

- [ ] Control the formats of coordinates in a consistent way in messages and reports both for DESIGNATE and TASK_A2G_DISPACHER. To if MGRS is selected in A2G settings, then DESIGNATE and TASK_A2G_DISPATCHER should work in MGRS mode for coordinates.

- [ ] The DESIGNATE menu system should also display coordinates if any in the chosen way in the A2G settings.

**2. Target Marking ...**

_I was thinking a "Mark Task Location on Map" option in "Join Task" or Status menu but not if it causes issues, I thought you'd be able to read a unit/area position for plotting tasks (if not tracking the data internally). Tactical Marking is "nice to have" but the overall MOOSE Framework and performance is more so._

I will look into creating a solution that allows to mark targets automatically on the map and an explanation. But you should realize something!!!! This should be part of the briefing! Maybe i can put these as part of a briefing report on the map when requested. There are several options and i will try to find the best as needed. My thinking was then it will save you time marking this stuff manually as the system can do it automatically. It does not necessarily need to provide the exact coordinates but rather mark the "location" or "area" and provide a descriptive text what the targets are the what the task is about... I will make that as part of the menu system.

But there is one thing you need to understand. The A2G Tasking and DESIGNATE are **not** reporting the unit positions on a real time basis. They are reporting the unit positions as a result of the detection!!! This is a big difference. So moving targets can be reported on the map, but not on a real time base but as part of a detection. The marking can potentially indicate the heading and speed as part of the detection to where the group is heading, as well as the detection time, threat level, group composition etc.

The level of "cheat" is up to the players to decide. By making this part of the menu system people can "cheat" up to preferences.

I can make it an option to "allow" the menu option to be displayed or not to mark targets on the map. So the mission designer is then in control.

**3. Lase targets and laser code ...**

No new requirement posed there, but I hope you've understood my clarifications. I will however change the text "Lase Target" to "Lase Targets with Random Laser Codes".

**4. Designate does more ...**

I will add all relevant available coordinate formats to the Detailed Task Report. Note that DESIGNATE also has a status report.

**5. Display time of reports ...**

I noticed you have an issue with reports only being displayed for 20 seconds.
What I will do is build a system in SETTINGS. Add options in A2G and A2A setting menus to select the timings for:
- Single Messages
- Summary Reports
- Detailed Reports

And I can then use this consistently in the logic and all gets variable and to user preferences.

FC

**Comment by Ramsay58** [comment]
**1. Task Planning ...**

 _I think you agree (as you haven't reacted) that the A2G settings need to:_

-  _Control the formats of coordinates in a consistent way in messages and reports both for DESIGNATE and TASK_A2G_DISPACHER. To if MGRS is selected in A2G settings, then DESIGNATE and TASK_A2G_DISPATCHER should work in MGRS mode for coordinates._

It's not so much that I agree but that I'm unsure the RL roll TASK_A2G_DISPACHER "single messages" represent, I also don't want this to be a personal "wish list".

I see TASK_A2G_DISPACHER as the HQ with *all* the information for tasking and "single messages" as CGI/AWACS directing the player to the Target Area so they know the players location and can use BRA.

-  _The DESIGNATE menu system should also display coordinates if any in the chosen way in the A2G settings._

I see DESIGNATE as the local JTAC/FAC - they know the targets location but not always the player, so use LL or MGRS.

My preference may be due to habit
- planning routes using way point/landmark BRA
- directing attacks using landmarks or map reference if I don't know where a player is.

However - in a town/forest, directing a helicopter attack or a FAC with eyes on both target and attacker -
 BRA is also appropriate.

So the answer is "it depends" and should be flexible.

**2. Target marking ...**

 _I will look into creating a solution that allows to mark targets automatically on the map and an explanation. But you should realize something!!!! This should be part of the briefing!_ 

The trouble with auto-generation is too many, irrelevant or overlapping Tactical Markers.

I have

- 6 infantry areas in Belorechensk (placeholder to represent the ground objective)
- 2 helicopter from Maykop AB supporting enemy forces (not part of the training/test)
- 5 ground targets/areas at Maykop AB (preventing capture) in my test mission.

So that's potentially 11 areas/tactical map marks. If flying a coop strike against the AB and we pre-sort targets:

- the detected infantry are irrelevant (but may have manpads, etc.)
- we might sort AB targets as North and South for ourselves
- then update/change tasks as the mission evolves

In a full blown mission - there'd be additional SAM's, friendly and enemy CAP, etc. responding as the AB is attacked, etc. but get in the way of a GBU-12 training mission to teach the basics.

_The A2G Tasking and DESIGNATE are not reporting the unit positions on a real time basis. They are reporting the unit positions as a result of the detection!!!_

Understood.
Unfortunately due to "Late activation" in MP causing problems, most of my ground forces are static.

**3. Lase targets and laser code ...**
Sounds good. I'm not sure if it's possible to smoke and simultaneously lase a target using DESIGNATE (smoke followed by lase perhaps?), I'll need to test some more before giving further feedback.

**4. Designate does more ...**

_I will add all relevant available coordinate formats to the Detailed Task Report. Note that DESIGNATE also has a status report._
Sounds good, will need to test the improvement before giving further feedback.

**5. Display time of reports ...**

_I noticed you have an issue with reports only being displayed for 20 seconds.
What I will do is build a system in SETTINGS. Add options in A2G and A2A setting menus to select the timings for:_

- _Single Messages_
- _Summary Reports_
- _Detailed Reports_
_And I can then use this consistently in the logic and all gets variable and to user preferences._

A great idea - to allow for the difference in reading a 1 line message and detailed report.

**Comment by FlightControl-Master** [comment]
@Ramsay58 what do you mean with RL roll in point 1?

**Comment by FlightControl-Master** [comment]
_I see DESIGNATE as the local JTAC/FAC_

DESIGNATE also reports to the HQ.
I want designate and tasking to be loosely coupled and working independently from each other.
But you have a point that reporting should be merely done using LL/MGRS.

**Comment by FlightControl-Master** [comment]
@Ramsay58 As a first action, I have reworked the DESIGNATE menu to work always in A2G mode. DESIGNATE is for A2G or G2G operations. I am keeping the options open for players to select the menu to presented in BR, LLDMS, LLDDM, MGRS or BULLS, depending on the settings menu. At least the BRA is now displayed as BR, according the A2G settings.

**Comment by Ramsay58** [comment]
@FlightControl-Master 
>what do you mean with RL roll in point 1?

TASK_A2G_DISPACHER "single messages" i.e. player BRA to a target area, are by necessity, an abstraction of complex real life C3 systems from a WWII static tactical brief to  modern, dynamic C4 GCI/AWACS "talk on's".

The BRA calls are a good abstraction for the prior route planning or AWACS/GCI talk-ons that might be done.

**Comment by FlightControl-Master** [comment]
Shall I make a setting to switch off these messages, or to make these messages being generated from potential named mark points placed over buildings, roads, forests, mountain tops?
I mean, as I am understanding, you would want these to be not showing up, correct?
Or at least not directly to target, but in reference to mark points, correct?
Just note that there are planes with very limited navigation equipment too...

**Comment by FlightControl-Master** [comment]
I thought routing messages would be useful. But yeah, I am no pilot.
We discussed it also in our group.
I wonder then how in WWII people were guided towards the targets.
I am working in these markpoints, so a task briefing to be displayed near the target area. NEAR THE,TARGET AREA, not on the target area. Also the briefing can contain altitude and heading/speed in terms of moving targets. And a time indication of the last detection.

Would a pilot ever get messages from HQ when executing an A2G task?
About target location etc?

**Comment by FlightControl-Master** [comment]
@Ramsay58 and all,

Added new functions that allow to configure the display timings of messages. Not all messages that are generated by moose have been converted into this system, but i plan to gradually do this.
There are basically 5 different message display types:
- Update Messages: these provide a short info (like route info, status update info etc.). Important is that these kind of messages can be switched off!
- Information Messages: these provide information to the pilot. Can't be switched off.
- Briefing Reports: The briefing reports are typically to be displayed a bit longer.
- Overview Reports: Reports that provide an overview. Can be displayed a bit longer than normal messages, but not too long. These reports may become large. So, they are not to be used for long-time displays. They are used to generate an "impression".
- Detailed Reports: Reports that provide details of a subject. These also may need to be viewed longer, as they may contain important information.

The SETTINGS menu has been added now with options to configure the display timings of these reports, both on a system level, as on a player level.

![screen_170912_113838](https://user-images.githubusercontent.com/13690775/30319617-a5d1b966-97b0-11e7-84ce-240e135c2532.jpg)

Each of these items have a submenu that allow you to select from a timing range:

![screen_170912_113840](https://user-images.githubusercontent.com/13690775/30319693-ce612fe2-97b0-11e7-9588-c5c0bf909979.jpg)

![screen_170912_113843](https://user-images.githubusercontent.com/13690775/30319700-d4533e86-97b0-11e7-8c23-20607ee4a639.jpg)

Each of the items have different sub menu timings... Depending on the kind of report.

The default values are in seconds:
```lua
      self:SetMessageTime( MESSAGE.Type.Briefing, 180 )
      self:SetMessageTime( MESSAGE.Type.Detailed, 60 )
      self:SetMessageTime( MESSAGE.Type.Information, 30 )
      self:SetMessageTime( MESSAGE.Type.Overview, 60 )
      self:SetMessageTime( MESSAGE.Type.Update, 15 )
```
So an update message will have a default of 15 seconds display. You can change that to 30 by using the menu or switch it off!!!

Messages will appear like this (i may still change this).

![screen_170912_114019](https://user-images.githubusercontent.com/13690775/30319858-4db26cca-97b1-11e7-8a2f-6107c79afb46.jpg)

So, when a message display timing menu item has been selected, you get a message of the change.
And messages that are using this system will act accordingly.

**With this system you can now disable the routing messages that are continuously displayed if you want!**

**And, you can now also update the duration of the reports that are being displayed from the tasking... If 20s is too slow, make it 2 minutes :-)**

That is the whole purpose why this stuff has been added.

Hope everyone likes it. Still working on it, but this will gradually be implemented everywhere.

**Comment by FlightControl-Master** [comment]
So, now as part of the tasking, before you join a task, you can place a marking at the target location from within the system. The task briefing will be displayed as part of the marking text:

![screen_170912_113213](https://user-images.githubusercontent.com/13690775/30320211-4c8e7086-97b2-11e7-8627-e359ba228c53.jpg)

Markings are triggered from the "join task" menu of the tasking...

Upon @Ramsay58 comments and feedback, the markings are not generated automatically, but must be individually requested as part of the briefing... This is work in progress...

![screen_170912_120458](https://user-images.githubusercontent.com/13690775/30320389-c0f91cbe-97b2-11e7-9e20-2e67dde8a9c8.jpg)


Markings are per group, so other players won't see those markings.

This is all WIP.

Interested in your feedback. I also tried not to show too much information at once. Also, The detailed information is anyway available at the reports. I should add the TASK.nnn ID to the marking...

interested in your feedback...

**Comment by Ramsay58** [comment]
@FlightControl-Master 
>interested in your feedback...

It looks very promising, unfortunately I wasn't able to test in [2.2.0 pre MOOSE build Timestamp: 20170908_1409](https://github.com/FlightControl-Master/MOOSE/releases)

Is 2.2.0 pre moose.lua likely to be updated anytime soon or should I setup a Eclipse dev environment?

**Comment by Ramsay58** [comment]
Unfortunately I was able to crash DCS 1.5.7.9459 trying to mark a task - I'll create a new bug report even though the Map Marking feature is WIP as it'll be easier to track/mark resolved.

**Comment by Ramsay58** [comment]
I only did a light test with the status message delay options as I didn't get to fly a full mission but it worked surprisingly well.

**Comment by FlightControl-Master** [comment]
> @Ramsay58 I only did a light test with the status message delay options as I didn't get to fly a full mission but it worked surprisingly well.

It is great. Thanks. The future is that the settings will be saved on the server or on your workstation, so that settings are persistent when you restart your mission and over different missions that use the MOOSE framework...

**Comment by FlightControl-Master** [comment]
> @Ramsay58 Unfortunately I was able to crash DCS 1.5.7.9459 trying to mark a task - I'll create a new bug report even though the Map Marking feature is WIP as it'll be easier to track/mark resolved.

Good. The reason is that target descriptions to be put into the marking box may become a bit too long, and will crash DCS. I've had many crashes like this already and still am searching for the root cause. It seems it has to do with the text not fitting into the boundary boxes of the marking. I will try to make the target descriptions a bit shorter and will cap the text to about 255 characters, which may be one of the causes.

**Comment by Mechanist11** [comment]
Did some short test today with A2G task dispatch and designate. Both using same detections.

- Smoke and illumination are not working with the current github build, only laser, and only once. It won't switch to another target once you destroyed the first. Also you cannot turn the laser on again wia menu.
- Task reports in A2G tasing are not working, no matter where I set the display time. They didn't show up.
- Autolase on and off is not working properly. You can turn it on by you can turn it off.

**Comment by Ramsay58** [comment]
Noticed something similar while testing the new Task Marker function in build 20170915_1940. Unfortunately there was a syntax error in my script when setting the laser codes, so couldn't submit a designate bug report with any certainty.

I need to drop a Gazelle into my test mission and run it at night to see what is going on with the FAC's, lasers. etc.

@Mechanist11 
Might be a good idea to flag 'Designate Marking' as separate issue to make it easier to track/close when resolved.

**Comment by FlightControl-Master** [comment]
> Task reports in A2G tasing are not working, no matter where I set the display time. They didn't show up.

Fixed this issue. My mistake. I should have seen this.

**Comment by FlightControl-Master** [comment]
> Smoke and illumination are not working with the current github build, only laser, and only once. It won't switch to another target once you destroyed the first. Also you cannot turn the laser on again wia menu.





> Autolase on and off is not working properly. You can turn it on by you can turn it off.

Both are fixed... My mistake. I made a mistake with the message timing implementation, which stopped messages from being displayed and crashed the designate logic.

**Comment by FlightControl-Master** [comment]
I will update the release moose.lua file, to correct these mistakes.

**Comment by FlightControl-Master** [comment]
Release updated.

**Comment by Mechanist11** [comment]
Autolase off still not works as intended.

Autolase set off, previously designated targes still lased, new one (F7) is not
![image](https://user-images.githubusercontent.com/19891419/30558536-189d09a4-9cb2-11e7-9139-68322d181795.png)

Also after lasing for 60s the lasing stops automatically but you have to stop it as well in radio command then initiate it again...

**Comment by FlightControl-Master** [comment]
This is by design

**Comment by FlightControl-Master** [comment]
Autolase = lase automatically targets

**Comment by Mechanist11** [comment]
I know, I turn if off before I took the screenshot, but it kept lasing the previously found targets. If you see F2 button it says turn it on. Now AREA 5 is a new target and it is not lased as it was appeared after i turned off the auto lase.

**Comment by FlightControl-Master** [comment]
okay, so you also wanna auto-turn-off the previously lased targets when you turn off auto-lase...


> 5. Display time of reports ...
> 
> I noticed you have an issue with reports only being displayed for 20 seconds.
> What I will do is build a system in SETTINGS. Add options in A2G and A2A setting menus to select the timings for:
> 
>     Single Messages
>     Summary Reports
>     Detailed Reports
>     And I can then use this consistently in the logic and all gets variable and to user preferences.
> 
> A great idea - to allow for the difference in reading a 1 line message and detailed report.

- [ ] This is done.

**Comment by FlightControl-Master** [comment]
> 3. Lase targets and laser code ...
> Sounds good. I'm not sure if it's possible to smoke and simultaneously lase a target using DESIGNATE (smoke followed by lase perhaps?), I'll need to test some more before giving further feedback.

I've raised issue #700 and moved this for enhancement in release 2.3.

**Comment by FlightControl-Master** [comment]
> 4. Designate does more ...
> 
> I will add all relevant available coordinate formats to the Detailed Task Report. Note that DESIGNATE also has a status report.
> Sounds good, will need to test the improvement before giving further feedback.

- [ ] I need to add in the detailed task report the LL and MGRS coordinates. Probably also show these coordinates when marking.

**Comment by FlightControl-Master** [comment]
> 2. Target Marking...
> But then again, I can't get all of the coordinates in the menu packed. It has an impact on performance ....
> 
> I was thinking a "Mark Task Location on Map" option in "Join Task" or Status menu but not if it causes issues, I thought you'd be able to read a unit/area position for plotting tasks (if not tracking the data internally). Tactical Marking is "nice to have" but the overall MOOSE Framework and performance is more so.

- [ ] Target marking is implemented in the A2G tasking menu. It is put under "Assign Target" menu. DESIGNATE won't have any marking function implemented now. Can be added in release 2.3. Target marking is considered closed and is related to issue #698 .

**Comment by FlightControl-Master** [comment]
@Ramsay58 and @Mechanist11 Can we have any other upcoming requirement or remark added as a new issue please?

I am closing this long issue thread, which was a very good thread! thanks for all the feedback!
